app/src/main/java/com/epfl/drawyourpath/userProfileCreation/UserGoalsInitFragment.kt

```diff
@@ -78,18 +89,27 @@ class UserGoalsInitFragment : Fragment(R.layout.fragment_user_goals_init) {
             }
 
             if (test1 && test2 && test3) {
-                database.setUserGoals(username, distanceGoal.toDouble(), timeGoal.toDouble(), nunberOfPathGoal)
-
-                val previousActivity = activity
-                if (previousActivity != null) {
-                    val fragManagement = previousActivity.supportFragmentManager.beginTransaction()
-                    val dataToEndProfileCreationFrag: Bundle = Bundle()
-                    //data to transmit to the UserGoalsInitFragment(username)
-                    dataToEndProfileCreationFrag.putString("userName", username)
-                    val endProfileCreationFrag = EndProfileCreationFragment()
-                    endProfileCreationFrag.arguments = dataToEndProfileCreationFrag
-                    fragManagement.replace(R.id.userGoalInitFragment, endProfileCreationFrag)
-                        .commit()
+                var userLog = FirebaseAuth.getUser()
+                if(isTest){
+                    userLog = MockAuth.MOCK_USER
+                }
+                if(userLog == null) {
+                    throw java.lang.Error("Any user is logged to the app.")
```

Comment:
Throwing in a fragment will crash the app. Here, if the user isn't logged, we don't want to terminate the app but to redirect to the login activity.

Reply:
Since, the user creation activity is launch just after the login, there must be a user logged in the app. If it's not case, there is an error in the logging protocol so we must launch an error.

Reply:
The SDK can logs the user out for a few reasons and `getUser()` will return `null` so it doesn't necesseraly means there is an error in the login activity. You are right that an error should be recorded but crashing the app isn't a user-friendly approach.

Reply:
Ok I will change for launching the login activity it's done